docs/proposals/pepxxx_build_isolation_passthrough.md

```diff
@@ -0,0 +1,123 @@
+# PEP ### - Build Isolation Passthrough
+
+| Resource            | Link                                                                |
+| ------------------- | ------------------------------------------------------------------- |
+| `PEP Link`          | `To Be Published`                                                   |
+| `DPO Discussion`    | [\[Build Isolation Passthrough\] Allowing a "package” to passthrough the build isolation](<https://discuss.python.org/t/build-isolation-passthrough-allowing-a-specific-package-to-always-passthrough-the-build-isolation/77284/>) |
+| `Github Repository` |                                                                     |
+
+## Abstract
+
+In modern software development, build isolation is a critical practice that ensures the consistency and reliability of
+builds. It involves creating an isolated environment where dependencies and build tools are controlled and do not
+interfere with the system or other projects. However, there are scenarios where certain tools or dependencies need to
+be accessed directly from the host environment.
+
+This proposal outlines a strategy to enhance the build isolation mechanism introduced by
+[PEP 518](http://peps.python.org/pep-0518/). Build isolation passthrough is a mechanism that allows
+selective access to host resources while maintaining the integrity of the isolated environment.
+This feature aims to address the challenges faced when a package depends on the ABI of another package, particularly in
+scenarios involving C/C++/Rust/CUDA/etc. dependencies.
+
+## Motivation
+
+With the adoption of `pyproject.toml` as specified in [PEP 518](http://peps.python.org/pep-0518/), build isolation has
+become a standard practice in Python packaging. However, there are cases where build isolation can be counterproductive,
+especially when dealing with packages that depend on the ABI of other packages. This proposal aims to provide a solution
```

Comment:
```suggestion
especially when dealing with packages that depend on the (unstable) ABI of other packages. This proposal aims to provide a solution
```

Added because the most widely used package with an ABI (NumPy) doesn't need this proposal, because the ABI is stable and both forward and backwards compatible (modulo a major version bump).

Reply:
I do like this idea; the lack of this feature is quite painful when depending on PyTorch. It probably needs a few more examples than only PyTorch in the motivation section, to show there's a real-world need beyond a single package.

Reply:
Building any extension on top of any Deep Learning is just a paiiin. 

But I could also reference the classic culprit of MPI with no stable ABI and no standard implementation. Same issue.